portfolio/src/main/webapp/script.js

```diff
@@ -67,5 +70,50 @@ function createListElement(name, text) {
  */
 function deleteCommentData() {
   const request = new Request('/delete-data', {method: POST});
-  fetch(request).then(result => getCommentData());
+  fetch(request).then(response => getCommentData());
+}
+
+/**
+ * Fetch user login status from LoginServlet.
+ */
+function getLoginStatus() {
+  fetch('/login').then(response => response.json()).then((userStatus) => {
+    if (userStatus.isLoggedIn) {
+      document.getElementById('comments-form-div').style.display = 'show';
+      document.getElementById('delete-button-div').style.display = 'show';
+      document.getElementById('handle-login').innerHTML =
+          userStatus.loginMessage;
+    } else {
+      document.getElementById('comments-form-div').style.display = 'none';
+      document.getElementById('delete-button-div').style.display = 'none';
+      const messageElement = document.createElement(COMMENT_CHILD_TAG);
+      messageElement.innerText = userStatus.loginMessage;
+      document.getElementById('handle-login').innerHTML =
+          messageElement.innerText;
+    }
+  });
+}
+
+/**
+ * Fetch URL of user-uploaded image from Blobstore servlet.
+ */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url')
+      .then((response) => {
+        console.log(response);
+        return response.text();
+      })
```

Comment:
collapse this into a single then since you're just calling text() on the response

Reply:
I think I need both- the first turns the url into text that I need to then use to set the form action. I tried condensing in a few different ways, including doing messageForm.action = imageUploadUrl.text() or just returning response.text() in a single then but they give me a "This site can't be reached" when I click submit :/